cfme/infrastructure/virtual_machines.py

```diff
@@ -590,6 +590,9 @@ def create(self, force_check_memory=False):
 
         def delete(self, cancel=False):
             self._nav_to_snapshot_mgmt()
+
+            self._click_tree_path(self.name or self.description)
```

Comment:
Is name/description display version dependent? randomly selected by UI?

Reply:
The snapshot in the tree is named differently when run against different providers. When running against vmware provider, the snapshot name is based on the name we gave it when it was created. When run against RHV provider, the snapshot does _not_ have a name, it has only description, and the snapshot name in the tree is based on that description.

Reply:
**Required**

Gotcha, thanks for explanation @apagac.

I'd like to see a simple change here to select `self.name` or `self.description` based on the type of provider found by `self.vm.provider` - using Provider.one_of() method to verify type.

That way we are selecting the appropriate field based on the provider type, and have tighter assertion over the display and behavior of the UI.  This potentially hides bugs or changes in the UI between versions.

Its done this way elsewhere in the file (actions property) but I don't think that's a good example, and would argue it should be selected based on provider type as well. The change to actions doesn't need to be made here.